### PR TITLE
ZCS-11169 : Log4j upgrade version to 2.17.1 changes

### PR DESCRIPTION
**Pull request by abhishekdeshmukh2209**
Issue :- Upgrade Log4j version to 2.17.1

### CODE REVIEW

File: src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java [hunk 1]
```diff
@@ -706,10 +715,27 @@ protected String getTestCaseID(Method method) {
...
+						FileAppender appender = FileAppender.newBuilder()
+				                .setName("file")
+				                .setLayout(PatternLayout.newBuilder()
+				                        .withPattern("%m%n")
+				                        .build())
+				                .withFileName(filename)
+				                .build();
+						//Appender a = new FileAppender(layout, filename, false);
+						Logger openQABasePackage = LogManager.getLogger(OpenQABasePackage);
+				        LoggerContext context = (LoggerContext) LogManager.getContext(false);
```
Comment (preshitaagrawal) [inline src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java]: please check indentation

Comment (abhishekdeshmukh2209) [reply src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java]: done please check.

File: src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java [hunk 2]
```diff
@@ -1575,7 +1610,8 @@ public static void main(String[] args) throws HarnessException, IOException {
 		String countTestsResult = "No results";
 		String executeTestsResult = "No results";
 
-		BasicConfigurator.configure();
+		Configurator.setLevel(LogManager.getRootLogger().getName(), Level.INFO);
+        Configurator.setLevel(LogManager.getLogger(ExecuteHarnessMain.class).getName(), Level.INFO);
```
Comment (preshitaagrawal) [inline src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java:1618]: incorrect indentation

File: conf/log4j.properties [hunk 3]
```diff
@@ -12,63 +12,73 @@
...
 # A2 uses PatternLayout.
-log4j.appender.A2.layout=org.apache.log4j.PatternLayout
-log4j.appender.A2.layout.ConversionPattern=%-4r [%t] %-5p %c %x - %m%n
+appender.A2.layout.type=PatternLayout
+appender.A2.layout.pattern=%-4r [%t] %-5p %c %x - %m%n
 
 # Test Case Step tracer
-#log4j.logger.testcase.trace=TRACE, A3
-#log4j.logger.com.zimbra.qa.selenium.framework.util.HarnessException=A3
+log4j.logger.testcase.trace=TRACE, A3
```
Comment (suchitgupta11) [inline conf/log4j.properties]: To define a logger below are the steps :
logger.{name}.name = fully classified class name/ package name/ category name/ unique name
logger.{name}.level = info/warn etc
logger.{name}.additivity = false
logger.{name}.appenderRef.{refname}.ref = {appenderRef}

File: conf/log4j.properties [hunk 4]
```diff
@@ -12,63 +12,73 @@
...
-log4j.rootLogger=INFO, A1, general
 
 # A1 is set to be a ConsoleAppender.
-log4j.appender.A1=org.apache.log4j.ConsoleAppender
+appender.A1.type=org.apache.log4j.ConsoleAppender
 
 # A1 uses PatternLayout.
-log4j.appender.A1.layout=org.apache.log4j.PatternLayout
-log4j.appender.A1.layout.ConversionPattern=%-4r [%t] %-5p %c %x - %m%n
+appender.A1.layout.type=PatternLayout
```
Comment (suchitgupta11) [inline conf/log4j.properties]: You also have to provide appender name like
appender.A1.name = STDOUT

File: conf/log4j.properties [hunk 5]
```diff
@@ -12,63 +12,73 @@
...
+appender.sleep.fileName=${outputDirectory}/sleep.txt
+appender.sleep.level=INFO
+appender.sleep.append=false
+appender.sleep.layout=org.apache.log4j.PatternLayout
+appender.sleep.layout.ConversionPattern=%-4r - %m%n
 
 # Set selenium to WARN
-log4j.logger.org.openqa=WARN
\ No newline at end of file
+rootLogger.level = WARN
```
Comment (suchitgupta11) [inline conf/log4j.properties]: also define a default appender to rootLogger. 
rootLogger.appenderRef.stdout.ref = STDOUT

**Review by suchitgupta11 — CHANGES_REQUESTED** [review]

File: conf/log4j.properties [hunk 6]
```diff
@@ -12,63 +12,121 @@
...
+logger.A2.level = info
+logger.A2.additivity = false
+logger.A2.appenderRef.file.ref = fileAppender
 
 # A2 is set to be a FileAppender.
-log4j.appender.A3=org.apache.log4j.FileAppender
-log4j.appender.A3.File=${outputDirectory}/testcase.txt
-log4j.appender.A3.MaxFileSize=100KB
-log4j.appender.A3.Append=false
+appender.A3.type=org.apache.log4j.RollingFileAppender
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Use RollingFile only as org.apache.log4j.RollingFileAppender is belongs to log4j-1.2.16

File: conf/log4j.properties [hunk 7]
```diff
@@ -12,63 +12,121 @@
...
+logger.A1.name = A1Console
+logger.A1.level = info
+logger.A1.additivity = false
+logger.A1.appenderRef.stdout.ref = console
 
 # A2 is set to be a FileAppender.
-log4j.appender.A2=org.apache.log4j.RollingFileAppender
-log4j.appender.A2.File=${outputDirectory}/selenium.txt
-log4j.appender.A2.MaxFileSize=100KB
+appender.A2.type=org.apache.log4j.RollingFileAppender
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Use RollingFile only as org.apache.log4j.RollingFileAppender is belongs to log4j-1.2.16

File: conf/log4j.properties [hunk 8]
```diff
@@ -12,63 +12,121 @@
...
-log4j.appender.sleep.layout.ConversionPattern=%-4r - %m%n
+appender.A3.layout.type=PatternLayout
+appender.A3.layout.pattern=m%n
+
+logger.A3.name = A3File
+logger.A3.level = info
+logger.A3.additivity = false
+logger.A3.appenderRef.stdout.ref = a3FileAppender
+
+appender.general.type=org.apache.log4j.RollingFileAppender
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Use RollingFile only as org.apache.log4j.RollingFileAppender is belongs to log4j-1.2.16

File: conf/log4j.properties [hunk 9]
```diff
@@ -12,63 +12,121 @@
...
+appender.general.fileName=${outputDirectory}/debug.txt
+appender.general.name=generalFileAppender
+appender.general.level=WARN
+appender.general.append=false
+appender.general.layout=org.apache.log4j.PatternLayout
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Use PatternLayout only as org.apache.log4j.PatternLayout is belongs to log4j-1.2.16, Please correct for other places too.

File: conf/log4j.properties [hunk 10]
```diff
@@ -12,63 +12,121 @@
...
+logger.A5.level = info
+logger.A5.additivity = false
+logger.A5.appenderRef.stdout.ref = A1
+logger.A5.appenderRef.file.ref = coverageFileAppender
+
+
+logger.A5.name = com.zimbra.qa.selenium.framework.util.CodeCoverage
+logger.A5.level = info
+logger.A5.additivity = false
+logger.A5.appenderRef.stdout.ref = A1
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Remove this line, as no appender defined as A1.

File: conf/log4j.properties [hunk 11]
```diff
@@ -12,63 +12,121 @@
...
+logger.general.name = generalFileLogger
+logger.general.level = info
+logger.general.additivity = false
+logger.general.appenderRef.file.ref = generalFileAppender
+
+
+logger.A5.name = com.zimbra.qa.selenium.framework.util.staf.StafAbstract
+logger.A5.level = info
+logger.A5.additivity = false
+logger.A5.appenderRef.stdout.ref = A1
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Remove this line, as no appender defined as A1

File: conf/log4j.properties [hunk 12]
```diff
@@ -12,63 +12,121 @@
...
+logger.A5.name = com.zimbra.qa.selenium.framework.util.CodeCoverage
+logger.A5.level = info
+logger.A5.additivity = false
+logger.A5.appenderRef.stdout.ref = A1
+logger.A5.appenderRef.file.ref = coverageFileAppender
+
+logger.A5.name = com.zimbra.qa.selenium.framework.core.ExecuteHarnessMain.MethodListener
+logger.A5.level = info
+logger.A5.additivity = false
+logger.A5.appenderRef.stdout.ref = A1
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Remove this line, as no appender defined as A1

File: conf/log4j.properties [hunk 13]
```diff
@@ -12,63 +12,121 @@
...
+logger.A5.appenderRef.file.ref = coverageFileAppender
+
+appender.coverage.type=RollingFile
+appender.coverage.fileName=${outputDirectory}/coverage.txt
+appender.coverage.name=coverageFileAppender
+appender.coverage.level=WARN
+appender.coverage.append=false
+appender.coverage.layout=org.apache.log4j.PatternLayout
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Use PatternLayout only as org.apache.log4j.PatternLayout is belongs to log4j-1.2.16.

File: conf/log4j.properties [hunk 14]
```diff
@@ -12,63 +12,121 @@
...
+logger.coverage.name = com.zimbra.qa.selenium.framework.util.SleepMetrics
+logger.coverage.level = info
+logger.coverage.additivity = false
+logger.coverage.appenderRef.file.ref = coverageFileAppender
+
+appender.sleep.type=RollingFile
+appender.sleep.fileName=${outputDirectory}/sleep.txt
+appender.sleep.level=INFO
+appender.sleep.append=false
+appender.sleep.layout=org.apache.log4j.PatternLayout
```
Comment (suchitgupta11) [inline conf/log4j.properties]: Use PatternLayout only as org.apache.log4j.PatternLayout is belongs to log4j-1.2.16,

**Review by suchitgupta11 — CHANGES_REQUESTED** [review]

**Review by suchitgupta11 — APPROVED** [review]

**Review by preshitaagrawal — APPROVED** [review]

File: ivy.xml [hunk 15]
```diff
@@ -24,7 +24,8 @@
 		<dependency org="jaxen" name="jaxen" rev="1.1-beta-6"/>
 		<dependency org="libidn" name="libidn" rev="1.24"/>
 		<dependency org="log4j" name="apache-log4j-extras" rev="1.0"/>
-		<dependency org="log4j" name="log4j" rev="1.2.16"/>
+		<dependency org="org.apache.logging.log4j" name="log4j-core" rev="2.17.1" />
+               <dependency org="org.apache.logging.log4j" name="log4j-api" rev="2.17.1" />
```
Comment (dasiyogesh) [inline ivy.xml]: indentation not right

Comment (abhishekdeshmukh2209) [reply ivy.xml]: corrected.

File: src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java [hunk 16]
```diff
@@ -775,10 +802,17 @@ public void afterInvocation(IInvokedMethod method, ITestResult result) {
 					appenders.remove(key);
 				}
 				if (a != null) {
-					OpenQALogger.removeAppender(a);
-					Logger.removeAppender(a);
-					a.close();
-					a = null;
+                                       Logger openQABasePackage = LogManager.getLogger(OpenQABasePackage);
```
Comment (dasiyogesh) [inline src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java:807]: indentation incorrect

File: src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java [hunk 17]
```diff
@@ -1276,9 +1310,10 @@ private boolean parseArgs(String arguments[]) throws ParseException, HarnessExce
 
 			// Processing log4j must come first so debugging can happen
 			if (cmd.hasOption('l')) {
-				PropertyConfigurator.configure(cmd.getOptionValue('l'));
+				Configurator.initialize(null, cmd.getOptionValue('l'));
 			} else {
-				BasicConfigurator.configure();
+			        Configurator.setLevel(LogManager.getRootLogger().getName(), Level.INFO);
```
Comment (dasiyogesh) [inline src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java:1318]: indentation incorrect

File: src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java [hunk 18]
```diff
@@ -1575,7 +1610,8 @@ public static void main(String[] args) throws HarnessException, IOException {
 		String countTestsResult = "No results";
 		String executeTestsResult = "No results";
 
-		BasicConfigurator.configure();
+		Configurator.setLevel(LogManager.getRootLogger().getName(), Level.INFO);
+                Configurator.setLevel(LogManager.getLogger(ExecuteHarnessMain.class).getName(), Level.INFO);
```
Comment (dasiyogesh) [inline src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java]: indentation incorrect

File: src/java/com/zimbra/qa/selenium/framework/util/ZimbraAccount.java [hunk 19]
```diff
@@ -1412,7 +1412,8 @@ public static void main(String[] args) throws HarnessException {
 		String domain = ConfigProperties.getStringProperty("server.host");
 
 		// Configure log4j using the basic configuration
-		BasicConfigurator.configure();
+		 Configurator.setLevel(LogManager.getRootLogger().getName(), org.apache.logging.log4j.Level.INFO);
```
Comment (dasiyogesh) [inline src/java/com/zimbra/qa/selenium/framework/util/ZimbraAccount.java]: indentation off

**Pull request closed without merging**